pkg/secretutils/secret_manager.go

```diff
@@ -74,34 +74,39 @@ func (sm *SecretManager) claimSecret(secret *corev1.Secret, owner client.Object,
 		needsUpdate = true
 	}
 	if owner != nil {
+		ownerUID := owner.GetUID()
 		ownerLog := log.WithValues(
 			"ownerKind", owner.GetObjectKind().GroupVersionKind().Kind,
 			"owner", owner.GetNamespace()+"/"+owner.GetName(),
-			"ownerUID", owner.GetUID())
-		if ownerIsController {
-			if !metav1.IsControlledBy(secret, owner) {
-				ownerLog.Info("setting secret controller reference")
-				if err := controllerutil.SetControllerReference(owner, secret, sm.client.Scheme()); err != nil {
-					return errors.Wrap(err, "failed to set secret controller reference")
-				}
-				needsUpdate = true
-			}
-		} else {
-			alreadyOwned := false
-			ownerUID := owner.GetUID()
-			for _, ref := range secret.GetOwnerReferences() {
-				if ref.UID == ownerUID {
+			"ownerUID", ownerUID)
+
+		alreadyOwned := false
+		needsCleanup := false
+		var updated []metav1.OwnerReference
+		for _, ref := range secret.GetOwnerReferences() {
+			if ref.UID == ownerUID {
+				// We used to add controller references to BMC
+				// secrets. This was wrong, update.
+				if ref.Controller != nil && *ref.Controller {
+					needsCleanup = true
+					continue // skip appending this item
+				} else {
 					alreadyOwned = true
-					break
 				}
 			}
-			if !alreadyOwned {
-				ownerLog.Info("setting secret owner reference")
-				if err := controllerutil.SetOwnerReference(owner, secret, sm.client.Scheme()); err != nil {
-					return errors.Wrap(err, "failed to set secret owner reference")
-				}
-				needsUpdate = true
+			updated = append(updated, ref)
+		}
+		if needsCleanup {
+			ownerLog.Info("removing the controller flag from the owner reference")
+			// Make sure to do this before SetOwnerReference below
+			secret.SetOwnerReferences(updated)
```

Comment:
It looks to me like `SetOwnerReference()` already does what we want (replaces the existing owner reference with a new one that does not set the controller flag, so it seems to me that we could just call it whenever `!alreadyOwned || needsCleanup` instead of having two different paths.